### PR TITLE
Use jobs check command for liveness probe check in airflow 2

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This PR proposes to remove the current use of python code for liveness probe check
commands in the Scheduler & Triggerer deployments

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Much nicer (when tests fixed)

File: chart/templates/_helpers.yaml [hunk 1]
```diff
@@ -607,6 +607,42 @@ Create the name of the cleanup service account to use
...
 {{- end }}
 
+{{define "scheduler_liveness_check_command"}}
+
+  {{- if semverCompare ">=2.0.0" .Values.airflowVersion }}
+  - sh
+  - -c
+  - |
+    CONNECTION_CHECK_MAX_COUNT=0 exec /entrypoint bash -c \
+    "export AIRFLOW__LOGGING__LOGGING_LEVEL=ERROR && export AIRFLOW__CORE__LOGGING_LEVEL=ERROR && \
```
Comment (jedcunningham) [inline chart/templates/_helpers.yaml]:
```suggestion
    "export AIRFLOW__LOGGING__LOGGING_LEVEL=ERROR && \
```

With 2.0.0+, this should be enough now.

File: chart/templates/_helpers.yaml [hunk 2]
```diff
@@ -607,6 +607,42 @@ Create the name of the cleanup service account to use
...
+    from airflow.utils.db import create_session
+    from airflow.utils.net import get_hostname
+    import sys
+    with create_session() as session:
+        job = session.query(SchedulerJob).filter_by(hostname=get_hostname()).order_by(
+            SchedulerJob.latest_heartbeat.desc()).limit(1).first()
+    sys.exit(0 if job.is_alive() else 1)"
+  {{- end }}
+{{- end }}
+{{define "triggerer_liveness_check_command"}}
```
Comment (jedcunningham) [inline chart/templates/_helpers.yaml:637]:
```suggestion

{{define "triggerer_liveness_check_command"}}
```

File: chart/templates/scheduler/scheduler-deployment.yaml [hunk 3]
```diff
@@ -165,9 +165,11 @@ spec:
             periodSeconds: {{ .Values.scheduler.livenessProbe.periodSeconds }}
             exec:
               command:
-{{- with .Values.scheduler.livenessProbe.command }}
-{{ toYaml . | indent 16 }}
-{{- end }}
+                  {{- if .Values.scheduler.livenessProbe.command }}
+                  {{ toYaml .Values.scheduler.livenessProbe.command  | indent 16 }}
+                  {{- else}}
+                  {{- include "scheduler_liveness_check_command" . | indent 16 }}
```
Comment (jedcunningham) [inline chart/templates/scheduler/scheduler-deployment.yaml]:
```suggestion
                  {{ toYaml .Values.scheduler.livenessProbe.command  | nindent 16 }}
                  {{- else}}
                  {{- include "scheduler_liveness_check_command" . | nindent 16 }}
```

We want this, no?

**Review by jedcunningham — CHANGES_REQUESTED** [review]

File: chart/templates/scheduler/scheduler-deployment.yaml (hunk 3 above)
Comment (ephraimbuddy) [reply chart/templates/scheduler/scheduler-deployment.yaml:171]: Not sure of this one

File: chart/templates/_helpers.yaml [hunk 4]
```diff
@@ -607,6 +607,43 @@ Create the name of the cleanup service account to use
...
 
+{{define "scheduler_liveness_check_command"}}
+
+  {{- if semverCompare ">=2.0.0" .Values.airflowVersion }}
+  - sh
+  - -c
+  - |
+    CONNECTION_CHECK_MAX_COUNT=0 exec /entrypoint bash -c \
+    "export AIRFLOW__LOGGING__LOGGING_LEVEL=ERROR && \
+    airflow jobs check --job-type SchedulerJob --hostname $(hostname)"
```
Comment (jedcunningham) [inline chart/templates/_helpers.yaml]:
```suggestion
    CONNECTION_CHECK_MAX_COUNT=0 AIRFLOW__LOGGING__LOGGING_LEVEL=ERROR \
    exec /entrypoint bash -c \
    "airflow jobs check --job-type SchedulerJob --hostname $(hostname)"
```
This is probably better...

Also, I don't think we need nested sh/bash, does this work?:
```suggestion
    CONNECTION_CHECK_MAX_COUNT=0 AIRFLOW__LOGGING__LOGGING_LEVEL=ERROR \
    exec /entrypoint airflow jobs check --job-type SchedulerJob --hostname $(hostname)
```

**Review by jedcunningham — APPROVED** [review]

File: chart/templates/_helpers.yaml (hunk 4 above)
Comment (ephraimbuddy) [reply chart/templates/_helpers.yaml]: Yeah. Worked

**Pull request merged**